File: app/src/main/java/nl/rsdt/japp/jotial/maps/wrapper/google/GoogleUiSettings.java
package nl.rsdt.japp.jotial.maps.wrapper.google;

import org.osmdroid.views.MapView;

import nl.rsdt.japp.jotial.maps.wrapper.IUiSettings;

/**
 * Created by mattijn on 09/08/17.
 */

public class GoogleUiSettings implements IUiSettings{
    private final com.google.android.gms.maps.UiSettings googleUiSettings;

    public GoogleUiSettings(com.google.android.gms.maps.UiSettings uiSettings) {
        googleUiSettings = uiSettings;
    }

    @Override
    public void setAllGesturesEnabled(boolean allesturesEnabled) {
        googleUiSettings.setAllGesturesEnabled(allesturesEnabled);
    }

    @Override
    public void setCompassEnabled(boolean compassEnabled) {
        googleUiSettings.setCompassEnabled(compassEnabled);
    }

    @Override
    public void setZoomControlsEnabled(boolean zoomControlsEnabled) {
        googleUiSettings.setZoomControlsEnabled(zoomControlsEnabled);
    }

    @Override
    public void setIndoorLevelPickerEnabled(boolean indoorLevelPickerEnabled) {
        googleUiSettings.setIndoorLevelPickerEnabled(indoorLevelPickerEnabled);
    }

    @Override
    public void setMapToolbarEnabled(boolean mapToolbarEnabled) {
        googleUiSettings.setMapToolbarEnabled(mapToolbarEnabled);
    }
}

File: app/src/main/java/nl/rsdt/japp/jotial/maps/wrapper/google/GoogleUiSettings.kt
package nl.rsdt.japp.jotial.maps.wrapper.google

import nl.rsdt.japp.jotial.maps.wrapper.IUiSettings

/**
 * Created by mattijn on 09/08/17.
 */

class GoogleUiSettings(private val googleUiSettings: com.google.android.gms.maps.UiSettings) : IUiSettings {

    override fun setAllGesturesEnabled(allesturesEnabled: Boolean) {
        googleUiSettings.setAllGesturesEnabled(allesturesEnabled)
    }

    override fun setCompassEnabled(compassEnabled: Boolean) {
        googleUiSettings.isCompassEnabled = compassEnabled
    }

    override fun setZoomControlsEnabled(zoomControlsEnabled: Boolean) {
        googleUiSettings.isZoomControlsEnabled = zoomControlsEnabled
    }

    override fun setIndoorLevelPickerEnabled(indoorLevelPickerEnabled: Boolean) {
        googleUiSettings.isIndoorLevelPickerEnabled = indoorLevelPickerEnabled
    }

    override fun setMapToolbarEnabled(mapToolbarEnabled: Boolean) {
        googleUiSettings.isMapToolbarEnabled = mapToolbarEnabled
    }
}
